### ISSUE TITLE
Distribution for Ubuntu users

### ISSUE THREAD

**Issue by aadcg** [atlas-engineer/nyxt#3048]
Nyxt would like to see itself available on Ubuntu. Nonetheless, it's hard to commit to such a task since, besides having other priorities, none of the team members either uses an Ubuntu flavored distribution or has experience packaging in it.

My understanding is that to have new software added to Ubuntu's repositories takes a fair bit of politics too. The first step would be to have a PPA. 

**Dear fellow community members, are you a Nyxt and Ubuntu user with enough experience to create a Nyxt PPA?** 

------

In the meantime, we may consider providing a `.deb` package again. @hgluka uses a Ubuntu flavored distribution, which makes it easier for us to have someone who can easily test it and access its robustness. 

On how to achieve the `.deb`, I'd suggest moving away from the strategy previously used and to focus on generating it via `guix pack`. It is proves to be good enough, it's would greatly simplify the work of maintaining the recipe that generates it.  

**Dear fellow community members, would you like us to distribute a `.deb`?** 

-----

The piece of information that triggers this discussion is that I found that the download numbers of `.deb`s (back in the end when we distributed them) was significant. Nyxt, while still being aimed at medium to advanced users, has been consistently decreasing the level of expertise needed to use it. Therefore I think it's safe to say that widening the target public will not backfire.

**Comment by fedreg-bn** [comment]
I would prefer a `.deb` over flatpak if you find an easy way to package it up,  but not a deal breaker or anything.   Not a huge fan of flatpak but it's perfectly usable.

(happy to help test out `.deb` builds if that helps)

**Comment by Ambrevar** [comment]
We used to provide debs and we have a framework to build them.  See https://github.com/atlas-engineer/build-script-archive.

**Comment by Ambrevar** [comment]
Because it provides very little benefit, only maybe that it "registers" the package so that DPKG can know that it exists, but then:
- it needs roots access
- it cannot be installed wherever the user wants.
- it's super heavy since it includes all recursive dependencies and does not reuse the Debian packages.

It's a big step backwards compared to our previous .deb generation (which worked really well).

**Comment by jmercouris** [comment]
I thought our previous .deb generation did not work due to packaging issues with the latest version of WebKitGTK+?

**Comment by Ambrevar** [comment]
No, it always worked, it's just that the .deb only worked on Ubuntu 20.** something, and not on older / newer Ubuntu.
Basically the .deb works on the Ubuntu that's used by the CI.

This is a limitation of non-functional OSes, nothing we can do about it.
One option of course is to generate multiple .debs, which is easy since the CI has multiple Ubuntu versions available.

**Comment by jmercouris** [comment]
Right, I forgot, they had deprecated a SSL library that we needed. This will not be the last time they break our package. In this case, the Guix package for generation has more benefits than our previous approach. What if we provided a snap (which is basically a flatpak) for Ubuntu, this could work easier perhaps?

**Comment by Ambrevar** [comment]
Maybe, but then that's a lot of packaging work on our behalf.

Providing a PPA, then hoping it gets merged upstream might be a more constructive approach.

**Comment by jmercouris** [comment]
Acknowledged!

**Comment by tecfu** [comment]
Would love to see this solutioned with an AppImage instead of a Flatpak. Curious about the existing preference for Flatpak here.

**Comment by tecfu** [comment]
@aadcg Running an AppImage doesn't require any additional installs. This makes evaluating the software much more accessible: A user simply has to: 

- Download nyxt.AppImage
- `chmod +x nyxt.AppImage`
- `./nyxt.AppImage`

With Flatpak, a user needs to to install yet another package manager on their system. Some people have no desire to do this, and it creates an extra step that makes trying out nyxt less likely.

**Comment by kurokirasama** [comment]
late to the party but yes, a deb file would be awesome

**Comment by bubbleguuum** [comment]
It's curious nobody made a .deb package for Debian/Ubuntu at this point. It's not rocket science to compile and package Nyxt.
Maybe asking on relevant Debian mailing-list (if any) may help ?

**Comment by aadcg** [comment]
Conclusion: interested parties should create a Nyxt PPA and eventually try to get it accepted in the package manager mainline.
